src/hotspot/share/opto/compile.cpp

```diff
@@ -4215,18 +4210,18 @@ void Compile::print_inlining_commit() {
   assert(print_inlining() || print_intrinsics(), "PrintInlining off?");
   // Transfer the message from _print_inlining_stream to the current
   // _print_inlining_list buffer and clear _print_inlining_stream.
-  _print_inlining_list->at(_print_inlining_idx).ss()->write(_print_inlining_stream->base(), _print_inlining_stream->size());
+  _print_inlining_list->at(_print_inlining_idx)->ss()->write(_print_inlining_stream->base(), _print_inlining_stream->size());
   print_inlining_reset();
 }
 
 void Compile::print_inlining_push() {
   // Add new buffer to the _print_inlining_list at current position
   _print_inlining_idx++;
-  _print_inlining_list->insert_before(_print_inlining_idx, PrintInliningBuffer());
+  _print_inlining_list->insert_before(_print_inlining_idx, new PrintInliningBuffer());
 }
 
 Compile::PrintInliningBuffer& Compile::print_inlining_current() {
-  return _print_inlining_list->at(_print_inlining_idx);
+  return *(_print_inlining_list->at(_print_inlining_idx));
```

Comment:
Maybe just change the return to `Compile::PrintInliningBuffer*` and update callers accordingly (there are only a few).